### ISSUE TITLE
Memory-leak in Java Process that executes Test Sutes/CURA/chronos_exam

### ISSUE THREAD

**Issue by kazurayam** [kazurayam/VisualTestingByImagemagickInKatalonStudio#13]
I created a branch named [MemoryUsageResearch]( https://github.com/kazurayam/VisualTestingInKatalonStudio/tree/MemoryUsageResearch)

# Setup monitoring tools
1. Install "Java VisutalVM". You can download it from https://visualvm.github.io/download.html 

# How to reproduce the problem
1. Specify `-Xmx10g` in `katalong.exe`
2. Clone out the MemoryUsageResearch branch
3. open `Test Suites/CURA/Execute_chronos - Debug` and click "execute"
4. Execute the Test Suite Collection 2 twice or more.
5. Pay attention how the Heap size moves when the Test Suite `CURA/chronos_exam` runs

<img width="1440" alt="スクリーンショット 2020-06-26 10 23 03" src="https://user-images.githubusercontent.com/4467417/85814993-434e2080-b7a2-11ea-9988-1e38dbe118f3.png">

You can see the Java process occupies 7giga-bytes of memory. The Java Heap size occuipes 7 Gbytes. Occasionally I found that the zone painted with lightblue goes up to 6 Gbytes; this means that the Java Garbage Collection can not free that amount of Heap. Unfreeable Heap? Why do we have that?

**Comment by kazurayam** [comment]
https://stackoverflow.com/questions/38190585/why-does-this-java-process-not-release-memory

>The memory reported in the System Monitor is all memory used by the process, not just Java heap. This memory includes the:

>Executable files of the VM itself, and the libraries it has loaded
>- Working space for the VM as a process for use in such things as the Hotspot compiler, GC, IO buffers, screen and graphic buffers, reading VM files and the like
>- Java Heap and other reportable memory structures

>In your case, 10MB of the process is being used to store the Java stack and the Java objects. The other 90MB is the Java program itself and the memory internal to the VM.

**Comment by kazurayam** [comment]
It seems that the some classes in the Materials library called by the Test Suite `CURA/chronos_exam` has a problem. It reads/writes large PNG images files using `javax.imageio.ImageIO.read(File)` and `javax.imageio.ImageIO.write(File)` like this:

```
try {
    readImage = ImageIO.read(new File("path to a PNG file"))
}
```

Those static methods might be problematic. It is likely that the memory for ImageIO is not promptly freed. I should rather write like this:

```
try {
    FileInputStream in = new FileInputStream(new File("file path))
    BufferedImage bi = ImageIO.read(in)
    in.close()
}
```

Please note that the application closes the InputStream as soon as it is done. By doing this, I guess the memory for ImageIO buffer will be promptly freed.

**Comment by kazurayam** [comment]
I have upgraded the Materials library to the version 0.77.0

https://github.com/kazurayam/Materials/releases/tag/0.77.0

I have changed the https://github.com/kazurayam/VisualTestingInKatalonStudio/tree/MemoryUsageResearch so that it uses the 0.77.0 of Materials. The following screenshot shows how the Java Heap moves:

<img width="1440" alt="スクリーンショット 2020-06-26 15 56 31" src="https://user-images.githubusercontent.com/4467417/85829807-625ea980-b7c6-11ea-94e1-5b72f940a1d8.png">

I could see

1. `-Xmx6g` is surely enough, smaller (e.g, -Xmx4g) would be fine
2. GC works fine. When I force GC manually, I could see the size of Stack goes down to nearly 0. This implies there is very little Non-freeable used memory like unclosed IO buffer.

**Comment by kazurayam** [comment]
I have upgraded the Materials library to the version 0.77.0

**Comment by kazurayam** [comment]
relased VisualTestingInKatalonStudio ver 1.19.0 which fixed #13
